### DISCUSSION TITLE
Spaced repetition algorithm?

### DISCUSSION THREAD

**Discussion by karol-brejna-i** [olmps/memo#100] [General (English)]
Could you shed some light on what is the algorithm for scheduling spaced repetitions in this app? 

Are you using something custom or something "known" (like SuperMemo)?

What are the modules/packages (in the code) responsible for that?

**Comment by matuella** [comment]
Sure!

We took inspiration in the SuperMemo's forgetting curve to make this first version of Memo, although we still know that there is a lot of room for improvements.

Just to clarify, we are using different terms from the classic flashcards here, so the usual Deck is called a Collection and its Memos are the Cards that belongs to a Deck. With this in mind, to evaluate the "next memos" that you are going to execute in any Collection, _we simply estimate an arbitrary value, ranging from 0 to 1, for what we call the memory recall, pertaining to (one or more) executions of a single Memo_.

The "code" that is responsible directly for this is the `MemoryRecallServices`, but more specifically the [`_evaluateMemoRecall`]( https://github.com/olmps/memo/blob/main/lib/domain/isolated_services/memory_recall_services.dart#L100) function. It's pretty well documented, so you can also take a look at that explanation. Other relevant services that are related to this are the [`CollectionServices`](https://github.com/olmps/memo/blob/main/lib/domain/services/collection_services.dart) and the [`ExecutionServices`](https://github.com/olmps/memo/blob/main/lib/domain/services/execution_services.dart), both which are directly related to the act of executing and displaying the memory recall inside the application.

I think the docs on all of these are straightforward but, just like I said, we are looking to improve this solution, as we are not experts on the area.

Hope that helps.
